tests/components/broadlink/test_config_flow.py

```diff
@@ -0,0 +1,748 @@
+    assert result["data"]["mac"] == device.mac
+    assert result["data"]["type"] == device.devtype
+
+    assert mock_api.auth.call_count == 1
+    assert mock_discover.call_count == 1
+
+
+async def test_flow_import_already_in_progress(hass):
+    """Test we do not import more than one flow per device."""
+    device = get_device("Living Room")
+    data = {"host": device.host}
+
+    with patch("broadlink.discover", return_value=[device.get_mock_api()]):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=data
+        )
+
+    with patch("broadlink.discover", return_value=[device.get_mock_api()]):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=data
+        )
+
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_in_progress"
+
+
+async def test_flow_import_host_already_configured(hass):
+    """Test we do not import a host that is already configured."""
+    device = get_device("Living Room")
+    mock_entry = device.get_mock_entry()
+    mock_entry.add_to_hass(hass)
+    mock_api = device.get_mock_api()
+
+    with patch("broadlink.discover", return_value=[mock_api]):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={"host": device.host},
+        )
+
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
+
+
+async def test_flow_import_mac_already_configured(hass):
+    """Test we do not import more than one config entry per device.
+
+    We need to abort the flow and update the existing entry.
+    """
+    device = get_device("Living Room")
+    mock_entry = device.get_mock_entry()
+    mock_entry.add_to_hass(hass)
+
+    device.host = "192.168.1.16"
+    mock_api = device.get_mock_api()
+
+    with patch("broadlink.discover", return_value=[mock_api]):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={"host": device.host},
+        )
+
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
+
+    assert mock_entry.data["host"] == device.host
+    assert mock_entry.data["mac"] == device.mac
+    assert mock_entry.data["type"] == device.devtype
+    assert mock_api.auth.call_count == 0
+
+
+async def test_flow_import_device_not_found(hass):
+    """Test we handle a device not found in the import step."""
+    with patch("broadlink.discover", return_value=[]):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={"host": "192.168.1.32"},
+        )
+
+    assert result["type"] == "abort"
+    assert result["reason"] == "cannot_connect"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)